### ISSUE TITLE
Large microSD's in RPi Zero W fail to boot due to fsck, when image small?

### ISSUE THREAD

**Issue by holta** [iiab/iiab#723]
Due to "fsck.repair=yes rootwait" in /boot/cmdline.txt ?

@tim-moody theory: writing small images to large (e.g. 128GB) microSD causes fsck (checkdisk) to time out on RPi Zero W, failing to boot almost every time (when it gives up after several minutes).

@georgejhunt wonders if we can increase the timeout?

**Comment by tim-moody** [comment]
This is all anecdotal. A large (say 128G) card on rpi W can fail to boot.

This can happen if there is a large amount of space to resize and also if fsck.repair needs to run.

**Comment by jvonau** [comment]
Looking at a [boot sequence from a rpi](https://raspberrypi.stackexchange.com/questions/78099/how-can-i-lower-my-boot-time-more) it appears that the resizing service will run much later than systemd-fsck-root.service that does the root file system check. Just a thought, perhaps iiab-rpi-root-resize.service could employ a Before=systemd-fsck-root.service line to have the resizing take place before the file system integrity check is preformed?

**Comment by tim-moody** [comment]
worth a try, but I think the biggest problem is how long even the fsck service can take if the card is large.

**Comment by holta** [comment]
@jvonau writes: "Tim Looking at the long fsck check failure on boot, perhaps sd-min should do the tune2fs before running e2fsck to prevent having the image contain what might be considered a dirty filesystem?"

**Comment by holta** [comment]
@tim-moody is this failure (caused by a slow fsck / disk check) at all related to yours?

![img_20180614_165337943 2pb](https://user-images.githubusercontent.com/2458907/41438234-9adb5ba8-6ff4-11e8-81ab-68e3c065aac1.jpg)

**Comment by holta** [comment]
RPi Zero W's very often prevent themselves from booting, due to the problem shown in the above screenshot &mdash; no doubt triggered by these slow/cautious disk checks that happen on every 3rd or 4th boot &mdash; causing very serious problems when you have no screen or keyboard (as is normal with Zero W's, hence this is very serious!)

Pasted in explicitly here:
```
[ TIME ] Timed out waiting for device dev-disk-by\x2dpartuuid-36d137ba\x2d01.device.
[DEPEND] Dependency failed for File System Check on /dev/disk/by-partuuid/36d137ba-01.
[DEPEND] Dependency failed for /boot.
[DEPEND] Dependency failed for Local File Systems.
[DEPEND] Dependency failed for Mark the need to relabel after reboot.
[ TIME ] Timed out waiting for device dev-serial1.device.
```
CLARIF: this happens even if IIAB contains very few services at all (e.g. Calibre, Kiwix, MediaWiki and nothing else!)

**Comment by jvonau** [comment]
Think this was solved in #2465

**Comment by holta** [comment]
> Think this was solved in #2465

Let's hope so.

(@tim-moody please speak up if there's more to this?)
